nixos/lib/testing/macos-host.nix

```
@@ -0,0 +1,17 @@
+{ config, lib, hostPkgs, ... }:
+
+{
+  config = lib.mkIf hostPkgs.stdenv.hostPlatform.isDarwin {
+    defaults = {
+      virtualisation.host.pkgs = hostPkgs;
+
+      # Mounting the whole nix store from the host is the default because
+      # this makes rebuilds of the VMs much faster and the VMs smaller.
+      # However, on macOS it currently leads to many crashing guest processes.
+      virtualisation.useNixStoreImage = true;
+      # Without this, the store image uses "raw" instead of "qcow2" and
+      # that throws some errors on macOS on startup of the VM
+      virtualisation.writableStore = true;
```

```suggestion
      # TODO: remove when `raw` errors are fixed.
      virtualisation.writableStore = true;
```

To record the intent. Maybe also for the line above?